integration/generic-tests/tftp_test.go

```diff
@@ -0,0 +1,469 @@
+// Copyright 2018-2025 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !race
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hugelgupf/vmtest/qemu"
+	"github.com/hugelgupf/vmtest/qemu/qnetwork"
+	"github.com/hugelgupf/vmtest/scriptvm"
+	"github.com/u-root/mkuimage/uimage"
+)
+
+func tftpVM(t *testing.T, name, script string, net *qnetwork.InterVM, mods ...uimage.Modifier) *qemu.VM {
+	fixedMods := []uimage.Modifier{
+		uimage.WithBusyboxCommands(
+			"github.com/u-root/u-root/cmds/core/cat",
+			"github.com/u-root/u-root/cmds/core/echo",
+			"github.com/u-root/u-root/cmds/core/grep",
+			"github.com/u-root/u-root/cmds/core/ip",
+			"github.com/u-root/u-root/cmds/core/mkdir",
+			"github.com/u-root/u-root/cmds/core/shasum",
+			"github.com/u-root/u-root/cmds/core/sleep",
+		),
+		uimage.WithCoveredCommands(
+			"github.com/u-root/u-root/cmds/exp/tftp",
+			"github.com/u-root/u-root/cmds/exp/tftpd",
+		),
+	}
+
+	return scriptvm.Start(t, name, script,
+		scriptvm.WithUimage(append(fixedMods, mods...)...),
+		scriptvm.WithQEMUFn(
+			qemu.WithVMTimeout(2*time.Minute),
+			net.NewVM(),
+		),
+	)
+}
+
+func TestTFTPTransfer(t *testing.T) {
+	net := qnetwork.NewInterVM()
+
+	// Prepare test files for VMs
+	testFilesForServer := uimage.WithFiles(
+		"testdata/tftp/small.txt:/small.txt",
+		"testdata/tftp/large.txt:/large.txt",
+		"testdata/tftp/binary.dat:/binary.dat",
+		"testdata/tftp/file_a.txt:/file_a.txt",
+		"testdata/tftp/file_b.txt:/file_b.txt",
+		"testdata/tftp/file_c.txt:/file_c.txt",
+	)
+
+	testFilesForClient := uimage.WithFiles(
+		"testdata/tftp/small.txt:/upload_small.txt",
+		"testdata/tftp/large.txt:/upload_large.txt",
+		"testdata/tftp/binary.dat:/upload_binary.dat",
+		"testdata/tftp/file_a.txt:/upload_file_a.txt",
+		"testdata/tftp/file_b.txt:/upload_file_b.txt",
+		"testdata/tftp/file_c.txt:/upload_file_c.txt",
+	)
+
+	// Expected SHA1 hashes of the test files
+	// These are hardcoded for reliable testing
+	const (
+		smallFileHash  = "9470c442585479bfad86d9a0a0daf01779c020b3"
+		largeFileHash  = "379278b59fbb7ce439a17344c95be5afd335d540"
+		binaryFileHash = "b359ebb365d1abc5a927518dcdd42fc00f012523"
+		fileAHash      = "0f8735a9dfe23fcd2195db61ca8f96f3150c5f9b"
+		fileBHash      = "a7ac17275300cb68ca88742cc87904c290128557"
+		fileCHash      = "f99c3d0a167bb79fb9c6f7c38f85dd67b28c0558"
```

Comment:
It seems as if these constants are never used in the code. Why keep them?

Reply:
No, this file is not used. It was generated by `shasum` to get the hashes, and I think it is a good idea to keep it for reference if the test gets extended or modified. So you can easily check which hash belongs to which file.